app/services/vacancy_algolia_search_builder.rb

```diff
@@ -118,8 +119,10 @@ def build_location_filter
   end
 
   def build_search_replica
-    return nil if sort_by.blank?
-    self.search_replica = ['Vacancy', sort_by].reject(&:blank?).join('_')
+    # Returning nil means the search should use the main index, not a replica
```

Comment:
You shouldn't need to change this. I would maybe add a public method `sort_option` which returns the correct default `sort_by` value or the `sort_by` that was selected by the user, and change the value in the `select_tag` in _sorting_options.html.haml